text/0026-rest-api/rest-api.yaml

```diff
@@ -0,0 +1,895 @@
+            application/json:
+              schema:
+                $ref: '#/components/schemas/ErrorResponse'
+              examples:
+                default:
+                  $ref: '#/components/examples/can-not-find-data-response-example'
+  /messages/by-hash/is-referenced-by-milestone:
+    get:
+      tags:
+        - messages
+      summary: Check if a message is referenced by a milestone.
+      description: Check if a message is referenced by a milestone.
+      operationId: get-is-referenced-by-milestone
+      parameters:
+        - in: query
+          name: hashes
+          schema:
+            type: array
+            items:
+              type: string
+          required: true
+          description: Hash of the message.
+      responses:
+        "200":
+          description: successful operation
+          content:
+            application/json:
+              schema:
+                $ref: "#/components/schemas/GetIsReferencedByMilestoneResponse"
+              examples:
+                Example 1:
+                  $ref: '#/components/examples/post-is-referenced-by-milestone-response-example-1'
+        "400":
+          description: unsuccessful operation
+          content:
+            application/json:
+              schema:
+                $ref: '#/components/schemas/ErrorResponse'
+              examples:
+                default:
+                  $ref: '#/components/examples/invalid-data-provided-response-example'
+        "404":
+          description: unsuccessful operation
+          content:
+            application/json:
+              schema:
+                $ref: '#/components/schemas/ErrorResponse'
+              examples:
+                default:
+                  $ref: '#/components/examples/can-not-find-data-response-example'
+  /messages:
+    post:
+      tags:
+        - messages
+      summary: Submit a message.
+      description: Submit a message. The node takes care of missing fields and tries to build the message. On success, the message will be stored in the Tangle. This endpoint will return the hash of the message by the response headers.
+      operationId: post-messages
+      requestBody:
+        content:
+          application/json:
+            schema:
+              $ref: "#/components/schemas/PostMessagesParameters"
+        required: true
+      responses:
+        "201":
+          description: successful operation
+          headers:
+            Location:
+              schema:
+                $ref: '#/components/schemas/GetMessagesResponse'
+              description: Hash of the submitted message.
+        "400":
+          description: unsuccessful operation
+          content:
+            application/json:
+              schema:
+                $ref: '#/components/schemas/ErrorResponse'
+              examples:
+                default:
+                  $ref: '#/components/examples/invalid-data-provided-response-example'
+  /transaction-messages/is-confirmed:
```

Comment:
This route is exactly the same as `​/messages​/by-hash​/is-referenced-by-milestone`. Are the hashes perhaps supposed to be transaction hashes instead?

Reply:
> 1. I believe we should rename `/transaction-messages` to just `/transactions`. For example, routes like `/transaction-messages/by-address` include the word `messages` for no apparent reason.
> 2. post examples still use `post-` where they should be `get-`
> 3. Responses should return array of objects instead of objects where each sub object is keyed by the identifier of the particular entity.
> 4. I am not sure whether the `peers` count in the `/info` response is useful information.
> 5. The API does not describe any kind of authentication mechanism which I believe at the end we should have, perhaps something simple like a JWT.

2) fixed
3) is being discussed here https://github.com/iotaledger/protocol-rfcs/pull/27#discussion_r488505450
4) fixed
5) in the end we should have an authentication mechanism, do you think this can be done at later stage?

Regarding 1.

Let me first explain what the idea behind it was. Maybe we can also solve https://github.com/iotaledger/protocol-rfcs/pull/27#discussion_r486779003.

Since not all messages include a signed-message payload the idea was to have `GET /messages/` endpoints that do things that all messages have in common and have separate `GET /transaction-messages/` endpoints that only do things that are relevant to transaction-messages.

> Are the hashes perhaps supposed to be transaction hashes instead?

No, `GET /transaction-messages/` endpoints consume message hashes not transaction hashes.

Reply:
It's not useful since `GET /messages` could already do these. Users are responsible for the logic to extract/filter transaction imho. If they really want to search certain transaction payloads, we should provide `GET /transactions` with parameters of `transaction-id`, `message-id` or more instead.

I suggest maybe `GET /transaction-messages` could be removed for now since it's not necessary at the moment. Let other endpoints complete all the works it can do.

Reply:
Since we did remove the /transaction-messages endpoints I will close this conversation.